ui/round/src/view/user.ts

```diff
@@ -66,10 +68,36 @@ export function userHtml(ctrl: RoundController, player: Player, position: Positi
   );
 }
 
-const signalBars = (signal: number) => {
-  const bars = [];
+const bars = (signal: number) => {
+  const bars: VNode[] = [];
   for (let i = 1; i <= 4; i++) bars.push(h(i <= signal ? 'i' : 'i.off'));
-  return h('signal.q' + signal, bars);
+  return bars;
+};
+
+const playerLag = () => {
+  const ping = Math.round(wsAverageLag());
+  return !ping ? 0 : ping < 150 ? 4 : ping < 300 ? 3 : ping < 500 ? 2 : 1;
+};
+
+const signalBars = (signal: number, isPlayer: boolean = false) => {
+  return h(
+    'signal.q' + signal,
+    {
+      hook: {
+        insert(node) {
+          const el = node.elm as HTMLSpanElement;
+          if (isPlayer) {
+            setInterval(() => {
+              const signal = playerLag();
+              el.className = 'q' + signal;
+              node.children = bars(signal);
```

Comment:
??!! have you tried this?

Reply:
I had tested it, but I didn't notice that the bars didn't change
I'll change it now